airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/message/DestinationMessage.kt

```diff
@@ -285,7 +288,16 @@ data class EnrichedDestinationRecordAirbyteValue(
                     ),
                 Meta.COLUMN_NAME_AB_EXTRACTED_AT to
                     EnrichedAirbyteValue(
-                        IntegerValue(emittedAtMs),
+                        if (extractedAtAsTimestampWithTimezone) {
+                            TimestampWithTimezoneValue(
```

Comment:
nit: probably worth making this into a utility method + using the same code here and in DestinationRecordToAirbyteValueWithMeta